pkg/operator/ceph/csi/peermap/clientProfile.go

```diff
@@ -0,0 +1,162 @@
+}
+
+func createOrUpdateClientProfileMappingCR(clusterContext *clusterd.Context, clusterInfo *cephclient.ClusterInfo, clientProfileMapping *csiopv1a1.ClientProfileMapping) error {
+	existingClientProfileMapping := &csiopv1a1.ClientProfileMapping{}
+	request := types.NamespacedName{Name: clientProfileMapping.Name, Namespace: clusterInfo.Namespace}
+	err := clusterContext.Client.Get(clusterInfo.Context, request, existingClientProfileMapping)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			err := clusterContext.Client.Create(clusterInfo.Context, clientProfileMapping)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create clientProfileMappings %q", clientProfileMapping.Name)
+			}
+			logger.Infof("successfully created clientProfileMappings %q", clientProfileMapping.Name)
+			return nil
+		}
+		return errors.Wrapf(err, "failed to get existing clientProfileMappings %q", clientProfileMapping.Name)
+	}
+
+	if reflect.DeepEqual(existingClientProfileMapping.Spec, clientProfileMapping.Spec) {
+		logger.Debug("no change in client profile mappings. Skipping update of clientProfileMapping")
+		return nil
+	}
+
+	existingClientProfileMapping.Spec = clientProfileMapping.Spec
+	err = clusterContext.Client.Update(clusterInfo.Context, existingClientProfileMapping)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update existing clientProfileMappings %q", clientProfileMapping.Name)
+	}
+	logger.Infof("successfully updated clientProfileMappings CR %q", clientProfileMapping.Name)
+
+	return nil
+}
+
+func generateClientProfileMappingCR(clusterContext *clusterd.Context, clusterInfo *cephclient.ClusterInfo, pool *cephv1.CephBlockPool) (*csiopv1a1.ClientProfileMapping, error) {
+	mappings, err := getClusterPoolIDMap(clusterContext, clusterInfo, pool)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get peer pool ID mappings for the pool %q", pool.Name)
+	}
+
+	mappingsSpecList := []csiopv1a1.MappingsSepc{}
+	for _, mapping := range *mappings {
+		mappingsSpec := csiopv1a1.MappingsSepc{
+			BlockPoolIdMapping: map[string]string{},
+		}
+		remoteClusterID, localClusterID := getMapKV(mapping.ClusterIDMapping)
+		logger.Debugf("cluster ID mappings: %q:%q", remoteClusterID, localClusterID)
+		mappingsSpec.LocalClientProfile = localClusterID
+		mappingsSpec.RemoteClientProfile = remoteClusterID
+		for _, poolIDMappings := range mapping.RBDPoolIDMapping {
+			sortedPoolIDKeys := sortedKeys(poolIDMappings)
+			for _, poolIDKey := range sortedPoolIDKeys {
+				remotePoolID := poolIDKey
+				localPoolID := poolIDMappings[poolIDKey]
+				mappingsSpec.BlockPoolIdMapping[remotePoolID] = localPoolID
+				logger.Debugf("pool ID mappings: %s:%s", remotePoolID, localPoolID)
+				mappingsSpecList = append(mappingsSpecList, mappingsSpec)
+			}
+		}
+	}
+
+	blockOwnerDeletion := false
+	clientProfileMapping := &csiopv1a1.ClientProfileMapping{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pool.Name,
+			Namespace: clusterInfo.Namespace,
```

Comment:
This will create problem (its not allowed) as the CR will get created in operator namespace not in ceph cluster namespace.

Reply:
@Madhu-1 Is this comment still  valid? I can see that `cephconnection and clientprofile CRs are not using operator namespace in master branch.